### PR TITLE
add c-blosc/1.5.4

### PR DESCRIPTION
**Pull request by AndrewMeadows**
Specify library name and version:  **c-blosc/1.5.4**

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [X] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by AndrewMeadows** [comment]
Hrm... I see it has failed but I don't have access to the CI output.

**Comment by AndrewMeadows** [comment]
Why add a recipe for such an old version of c-blosc?  Because OpenVDB is stuck on c-blosc-1.5 and I'd like to add an openvdb recipe to conan-center-index.  This c-blosc/1.5.4 recipe is a dependency hurdle.

**Comment by uilianries** [comment]
Hi @AndrewMeadows , please consider https://github.com/AndrewMeadows/conan-center-index/pull/1

**Comment by AndrewMeadows** [comment]
I closed the [PR](https://github.com/AndrewMeadows/conan-center-index/pull/1) with suggested fixes but supplied something else which I hope will satisfy.

File: recipes/c-blosc/all/conanfile.py [hunk 1]
```diff
@@ -58,8 +61,10 @@ def requirements(self):
             self.requires.add("snappy/1.1.8")
         if self.options.with_zlib:
             self.requires.add("zlib/1.2.11")
-        if self.options.with_zstd:
-            self.requires.add("zstd/1.4.4")
+        # support for zstd was added in v1.11.1
+        if tools.Version(self.version) >= "1.11.1":
+            if self.options.with_zstd:
+                self.requires.add("zstd/1.4.4")
```
Comment (uilianries) [inline recipes/c-blosc/all/conanfile.py:67]:
```suggestion
        if self.options.get_safe("with_zstd"):
            self.requires.add("zstd/1.4.4")
```

File: recipes/c-blosc/all/conanfile.py [hunk 2]
```diff
@@ -85,7 +90,8 @@ def _configure_cmake(self):
         self._cmake.definitions["DEACTIVATE_LZ4"] = not self.options.with_lz4
         self._cmake.definitions["DEACTIVATE_SNAPPY"] = not self.options.with_snappy
         self._cmake.definitions["DEACTIVATE_ZLIB"] = not self.options.with_zlib
-        self._cmake.definitions["DEACTIVATE_ZSTD"] = not self.options.with_zstd
+        if tools.Version(self.version) >= "1.11.1":
+            self._cmake.definitions["DEACTIVATE_ZSTD"] = not self.options.with_zstd
```
Comment (uilianries) [inline recipes/c-blosc/all/conanfile.py:94]:
```suggestion
        self._cmake.definitions["DEACTIVATE_ZSTD"] = not self.options.get_safe("with_zstd", False)
```

**Comment by AndrewMeadows** [comment]
Closing this PR.

We were motivated to have c-blosc-1.5.4 packages because OpenVDB-7.0.0 claims problems were reported for higher versions of c-blosc, but that it is well tested on c-blosc-1.5.  However, our own experience found c-blosc-1.5.4 tests segfault (on Mac with clang-9, which is one of the build modes that failed for this PR), and we decided to brave higher versions of c-blosc on OpenVDB instead of digging into those failures.

The hope is: OpenVDB will Just Work on the latest and greatest c-blosc (which at this moment is 1.18.1).  Full speed ahead!

**Pull request closed without merging**